### PR TITLE
STORM-1221. Create a common interface for all Trident spout.

### PR DESCRIPTION
**Pull request by haohui**
This PR marks the trident spouts with a common interface. Such a common interface allows StormSQL to manage external data sources with a unified interface.

### CODE REVIEW

**Comment by Parth-Brahmbhatt** [comment]
+1.

**Comment by HeartSaVioR** [comment]
+1 as well.

**Comment by harshach** [comment]
+1

**Comment by hmcl** [comment]
+1

File: storm-core/src/jvm/storm/trident/TridentTopology.java [hunk 1]
```diff
@@ -127,7 +128,21 @@ public Stream newStream(String txId, IPartitionedTridentSpout spout) {
...
+        if (dataSource instanceof IBatchSpout) {
+            return newStream(txId, (IBatchSpout) dataSource);
+        } else if (dataSource instanceof ITridentSpout) {
+            return newStream(txId, (ITridentSpout) dataSource);
+        } else if (dataSource instanceof IPartitionedTridentSpout) {
+            return newStream(txId, (IPartitionedTridentSpout) dataSource);
+        } else if (dataSource instanceof IOpaquePartitionedTridentSpout) {
+            return newStream(txId, (IOpaquePartitionedTridentSpout) dataSource);
+        } else {
+            throw new UnsupportedOperationException();
```
Comment (hmcl) [inline storm-core/src/jvm/storm/trident/TridentTopology.java]: Perhaps you could add a meaningful message here. Something like "unsupported stream"

Comment (haohui) [inline storm-core/src/jvm/storm/trident/TridentTopology.java]: Addressed in the latest version. Thanks!

**Comment by harshach** [comment]
thanks @haohui  merged into master.

**Pull request merged**